From d6a39612929727373b0c3e1a42839f0cac96da86 Mon Sep 17 00:00:00 2001
From: hmz007 <hmz007@gmail.com>
Date: Mon, 5 Jul 2021 14:47:15 +0800
Subject: [PATCH 3/4] net: stmmac: Use phy_find_first() for invalid phy_addr

This make phy-handle optional, and without this property,
it enables automatic phy discovery.

Signed-off-by: hmz007 <hmz007@gmail.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index 3134f7e669f8..1ce6a3551891 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -1068,7 +1068,10 @@ static int stmmac_init_phy(struct net_device *dev)
 		int addr = priv->plat->phy_addr;
 		struct phy_device *phydev;
 
-		phydev = mdiobus_get_phy(priv->mii, addr);
+		if (addr == -1)
+			phydev = phy_find_first(priv->mii);
+		else
+			phydev = mdiobus_get_phy(priv->mii, addr);
 		if (!phydev) {
 			netdev_err(priv->dev, "no phy at addr %d\n", addr);
 			return -ENODEV;
-- 
2.31.0

